static/src/javascripts/projects/common/modules/commercial/sticky-top-banner.js

```diff
@@ -46,26 +48,17 @@ define([
     // Rubicon ads are loaded via DFP like all other ads, but they can
     // render themselves again at any time
     var newRubiconAdHeightPromise = new Promise(function (resolve) {
-        window.addEventListener('message', function (event) {
-            var data;
-            // other DFP events get caught by this listener, but if they're not json we don't want to parse them or use them
-            try {
-                data = JSON.parse(event.data);
-            } catch (e) {/**/}
-
-            if (data) {
-                var $iframe = getAdIframe();
-                var isRubiconAdEvent = data.type === 'set-ad-height';
-                var isEventForTopAdBanner = isRubiconAdEvent && data.value.id === $iframe[0].id;
-
-                if (isRubiconAdEvent && isEventForTopAdBanner) {
-                    fastdom.read(function () {
-                        var padding = parseInt($adBannerInner.css('padding-top'))
-                            + parseInt($adBannerInner.css('padding-bottom'));
-                        var clientHeight = parseInt(data.value.height) + padding;
-                        resolve(clientHeight);
-                    });
-                }
+        messenger.register('set-ad-height', function onSetAdHeight(data) {
+            var $iframe = getAdIframe();
```

Comment:
Which iframe do I grab if there are multiple instances of this advert on a page?

I only remember this hazily, but I'm sure that when I was thinking about writing a replacement to Google's Safeframes implementation, that this was an awkward problem. You can get the window of the origin frame, but not its actual container in the host page.

Instead - and I might be misremembering the matter entirely - what you had to do was have each frame send a ping to its host as it loaded, at which point the host would send each a message containing it's container's identifier. Future messages from the frame would contain this ID, so you could reliably pluck its container from the DOM and manipulate it as needs be.

But my memory is very foggy.